### ISSUE TITLE
google_checks.xml: false negative 2.3.1 Whitespace characters

### ISSUE THREAD

**Issue by Atharv3221** [checkstyle/checkstyle#21446]
Child Issue of #21428

I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

[2.3.1 Whitespace characters](https://google.github.io/styleguide/javaguide.html#s2.3.1-whitespace-characters)

<img width="991" height="162" alt="Image" src="https://github.com/user-attachments/assets/47489a1e-606e-4071-8852-753ea22a6ea2" />

Whitespaces: https://en.wikipedia.org/wiki/Whitespace_character

```java
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat Test.java
/** Raw whitespace U2000. */
public class Test {

  private static final String S = "line one";

  private static final String T =
      """
      line one
      """;
}
```

```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-14.1.0-all.jar -c /google_checks.xml Test.java
Starting audit...
Audit done.
```

Expected:
```
Starting audit...
[ERROR] /home/atharv/test/Test.java:4:35: Whitespace other than the ASCII space must be escaped.
[ERROR] /home/atharv/test/Test.java:7:10: Whitespace other than the ASCII space must be escaped. 
Audit done.
Checkstyle ends with 2 errors.
```

**Comment by Atharv3221** [comment]
@romani we can cover all non allowed whitespaces using [IllegalTokenText](https://checkstyle.org/checks/coding/illegaltokentext.html)

Example
```java
/** Raw whitespace U2000. */
public class Test {

  private static final String S = "line one";

  private static final String T =
      """
      line one
      """;
}
```

Using IllegalTokenText config
```
<module name="IllegalTokenText">
  <property name="id" value="whitespaceCharactersMustBeEscaped"/>
  <property name="tokens" value="STRING_LITERAL, CHAR_LITERAL, TEXT_BLOCK_CONTENT"/>
  <property name="format"
            value="[\x{2000}]"/>
  <property name="message"
            value="Whitespace other than the ASCII space must be escaped."/>
</module>
```

```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-14.1.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/atharv/test/Test.java:4:35: Whitespace other than the ASCII space must be escaped. [IllegalTokenText]
[ERROR] /home/atharv/test/Test.java:7:10: Whitespace other than the ASCII space must be escaped. [IllegalTokenText]
Audit done.
Checkstyle ends with 2 errors.
```

**Comment by NianJiuZst** [comment]
I am on it.

**Comment by Atharv3221** [comment]
@NianJiuZst please work on another issue, this is project related

**Comment by Atharv3221** [comment]
I am working on it

**Comment by romani** [comment]
Fix is merged
